tests/simapp/app.go

```diff
@@ -530,6 +531,28 @@ func (app *SimApp) RegisterAPIRoutes(apiSvr *api.Server, apiConfig config.APICon
 	}
 }
 
+// wrapAndSetPreBlocker sets the application's PreBlocker while still calling the originals.
+// This is needed to set custom PreBlock logic, but still allow things like x/upgrade to function.
+func (app *SimApp) wrapAndSetPreBlocker(pb sdk.PreBlocker) {
```

Comment:
Let's export this fn publicly? This is an operation all integrators have to roll on their own currently

Reply:
Yeah, that makes sense to me

Reply:
updated PR to reflect the change, let me know if all is good @nivasan1 @aljo242